pkgs/applications/audio/spotify-player/default.nix

```diff
@@ -30,31 +48,37 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs = [
     openssl
-    alsa-lib
     dbus
     fontconfig
-    libsixel
-  ];
+  ]
+    ++ lib.optionals withSixel [ libsixel ]
+    ++ lib.optionals (withAudioBackend == "alsa") [ alsa-lib ]
+    ++ lib.optionals (withAudioBackend == "pulseaudio") [ libpulseaudio ]
+    ++ lib.optionals (withAudioBackend == "rodio") [ alsa-lib ]
+    ++ lib.optionals (withAudioBackend == "portaudio") [ portaudio ]
+    ++ lib.optionals (withAudioBackend == "jackaudio") [ libjack2 ]
+    ++ lib.optionals (withAudioBackend == "rodiojack") [ alsa-lib libjack2 ]
+    ++ lib.optionals (withAudioBackend == "sdl") [ SDL2 ]
+    ++ lib.optionals (withAudioBackend == "gstreamer") [ gst_all_1.gstreamer gst_all_1.gst-devtools gst_all_1.gst-plugins-base gst_all_1.gst-plugins-good ];
 
   buildNoDefaultFeatures = true;
 
-  buildFeatures = [
-    "rodio-backend"
-    "media-control"
-    "image"
-    "lyric-finder"
-    "daemon"
-    "notify"
-    "streaming"
-    "sixel"
-  ];
+  buildFeatures = [ ]
+    ++ lib.optionals (withAudioBackend != "") [ "${withAudioBackend}-backend" ]
+    ++ lib.optionals withMediaControl [ "media-control" ]
+    ++ lib.optionals withImage [ "image" ]
+    ++ lib.optionals withLyrics [ "lyric-finder" ]
+    ++ lib.optionals withDaemon [ "daemon" ]
+    ++ lib.optionals withNotify [ "notify" ]
+    ++ lib.optionals withStreaming [ "streaming" ]
+    ++ lib.optionals withSixel [ "sixel" ];
 
   meta = with lib; {
```

Comment:
Let's update it to fit the convention of not using `with lib;` here.

Reply:
sure, whats the replacement for that?

Reply:
This should be good now. I can also expand `with lib.maintainers` but that feels unnecessarily verbose.